crates/orchestrator/src/network_spec/node.rs

```diff
@@ -0,0 +1,223 @@
+use configuration::shared::{
+    node::{EnvVar, NodeConfig},
+    resources::Resources,
+    types::{Arg, AssetLocation, Command, Image},
+};
+use multiaddr::Multiaddr;
+
+use crate::{
+    errors::OrchestratorError,
+    generators,
+    network::AddNodeOpts,
+    shared::types::{ChainDefaultContext, NodeAccounts, ParkedPort},
+};
+
+/// A node configuration, with fine-grained configuration options.
+#[derive(Debug, Clone)]
+pub struct NodeSpec {
```

Comment:
Im not sure if this can happen - but instead of adding `pub(crate)` for each var of the struct - can we just:
```suggestion
pub(crate) struct NodeSpec {
```
?

Reply:
If you add ```pub (crate)``` here, the type will only be exposed internally, while using pub here and ```pub (crate)``` per field make the field public and expose any external implemented traits method and also locally implemented public method (like below).

In the doc it will show as something like this:
```rust
pub struct NodeSpec {
    hidden
}
```

but you will have access to the method exposed